.github/workflows/pr-rtd-link.yml

```diff
@@ -0,0 +1,19 @@
+# The ReadTheDocs preview link is "hidden" within the GitHub "Checks"
+# interface. For users who don't know this, finding the preview link may be
+# very difficult or frustrating. This workflow makes the link more
+# findable by updating PR descriptions to include it.
+name: "Add ReadTheDocs preview link to PR descriptions"
+
+on:
+  pull_request_target:
+
+permissions:
+  pull-requests: "write"
```

Comment:
I could be totally misunderstanding this, but does this give PRs from anyone write permissions to `nsidc/earthaccess`?

Reply:
This gives the action permission to do things specifically to pull requests, for example add labels, comments, or in this case, edit the description of the PR. [Full list](https://docs.github.com/en/rest/authentication/permissions-required-for-github-apps?apiVersion=2022-11-28#repository-permissions-for-pull-requests)

Also, because the action uses `pull_request_target` event, it is restricted from seeing the code in the PR branch; it sees only the code in the target branch which we own. This means that the action that runs will be the one on the target branch and contributors can't, for example, change our GitHub Actions workflow YAML to do something malicious to all our pull requests.

Reply:
Okay, thanks for the extra context and clarification 

> This means that the action that runs will be the one on the target branch and contributors can't, for example, change our GitHub Actions workflow YAML to do something malicious to all our pull requests.

Great -- that's what I was wondering about